ls1intum/Artemis
# Rename modeling exercise server routes to follow REST guidelines

### Checklist

### Motivation and Context


We want to be consistent in naming our REST Endpoints. The changed endpoints didn't follow our REST Guidelines.

### Description

The changes reflects the table of REST routes https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=30016376
Changed routes in this PR:
- GET modeling-editor/{participationId} -> participations/{participationId}/latest-submission
- PUT modeling-submissions/{submissionId}/feedback -> modelling-submissions/{submissionId}/assessment
- PUT exercise/{exerciseId}/submission/{submissionId}/modelingExampleAssessment -> modeling-submissions/{submissionId}/example-assessment
- GET modeling-submissions/{submissionId}/result -> modeling-submissions/{submissionId}/assessment
- PUT modeling-exercises -> modeling-exercises/{exerciseId}

**Test Coverage**
Thanks to @madwau all the REST endpoints (Modeling assessment, submission and exercise)  are covered with tests, I adjusted the tests to target renamed endpoints and further improved the test coverage.
![Screenshot_2020-03-05 de tum in www1 artemis web rest](https://user-images.githubusercontent.com/17262376/75931622-bc350f80-5e75-11ea-9f7a-93c4edffb3c5.png)
![Screenshot_2020-03-05 de tum in www1 artemis web rest(1)](https://user-images.githubusercontent.com/17262376/75931630-c1925a00-5e75-11ea-8bcb-019731bbfaa5.png)
![Screenshot_2020-03-05 de tum in www1 artemis web rest(2)](https://user-images.githubusercontent.com/17262376/75932191-27cbac80-5e77-11ea-81da-df26d89a527e.png)

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create and start modeling exercise with example assessment
4. Check that as an instructor you can update it.
5. Participate and submit a solution as a student.
6. Assess this exercise as a tutor.
